src/interface_r/R/generics.R

```diff
@@ -21,3 +21,9 @@ fit_transform <- function(object, ...) {
 fit_predict <- function(object, ...) {
   UseMethod("fit_predict")
 }
+
+# Note that we are redefining `transform` generic method here to override `base::transform`'s signature.
+#' @export
+transform <- function(object, ...) {
```

Comment:
What does the H2O4GPU `transform()` method do?  I'm curious if we need this or if it's some sklearn artifact that's not necessary in R.

Reply:
It's usually for unsupervised algorithms but IMO it's the same as `predict()`. @navdeep-G What do you think?

Reply:
Ok, if it's just for sklearn and not 100% necessary, then I'd say let's leave it out for now.  We should ship the first version with the bare essentials so we don't end up wanting to remove stuff later. 

I am also thinking we should remove `fit_transform()` and `fit_predict()` and probably `score()`:

- It seems better to follow the standard R convention of separating fit from predict.  
- Do we have any other way to scoring our models right now other than `score()`?  If they use GPU methods, will the sklearn `score()` method work (i think it would just call sklearn to do that underneath)?  We have some options here:
  - We could expect people to use a third-party metrics package (e.g. [Metrics](https://cran.r-project.org/web/packages/Metrics/index.html) is pretty easy to use and comprehensive) to evaluate their models.  
  - We could provide our own built-in metrics functions like we do in H2O-3.
  - We could write a wrapper on top of *Metrics* to give a more integrated feel and return something similar to what we have in H2O-3.
   - Give them `score()` (though that's not very useful since it only provides a single metric, which is usually not the one you want actually -- it's accuracy for classification and R^2 for regression).

We could remove these here or in a follow-up PR.